src/UI/components/map/layers/drawerList.test.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { render } from '../../../test_config/render';
+import { screen, fireEvent } from '@testing-library/dom';
+import DrawerList from './drawerList';
+import { initialState } from '../../../state/mapSlice';
+
+describe('Drawer list components display and interaction testing', () => {
+  let testState;
+  beforeEach(() => {
+    testState = { map: JSON.parse(JSON.stringify(initialState)) };
+    testState.map.map_drawer = { open: true, overlays: true, baseMap: true };
+  });
+
+  describe('Overlays list is rendered correctly and interaction behaves as expected', () => {
+    it('when overlays list is open, it renders the correct number of overlays and their details', () => {
+      const overlays = [
+        {
+          name: 'testOverlay',
+          sourceLayer: 'testLayer',
+          sourceType: 'testType',
+        },
+      ];
+
+      render(
+        <DrawerList
+          sectionTitle="Overlays"
+          overlays={overlays}
+          sectionFlag="overlays"
+        />,
+        testState
+      );
+
+      const numOverlays = screen.getByTestId('overlaysListContainer').children
+        .length;
+      expect(numOverlays == overlays.length);
+    });
+
+    it('dispatches a toggle action to display the list of overlays', () => {
```

Comment:
Can we have another test for when the drawer is not open? Looks like that should set off two actions.